File: Binary Tree/tree.cpp
#include <iostream>
using namespace std;

struct node{
    int data;
    node* right;
    node* left;
};

class binaryTree{

    private:
        static node* root;

        void move(node* newnode, node* current, int val){
            if(newnode->data < current->data){
                    if(current->right == nullptr){
                        current->right = newnode;
                    }
                    else{
                        move(newnode, current->right, val);
                    }
                }
                else if(newnode->data > current->data){
                    if(current->left == nullptr){
                        current->left = newnode;
                    }
                    else{
                        move(newnode, current->left, val);
                    }
            }
        }

        node* search(int val, node* current){

            if(current->data < val){
                return current-> right;
            }
            if(current->data > val){
                return current-> left;
            }
            return nullptr;
        }

    public:

        binaryTree(){
            root = nullptr;
        }

        void add(int val){
            node* newnode = new node;
            newnode->data = val;
            newnode->right = nullptr; newnode->left = nullptr;
            if (root == nullptr){
                root = newnode;
            }
            else{
                node* current = root;
                move(newnode, current, val);
            }
        }

        

        void search(int val){
            node* current = search(val, root);

            if( current == nullptr ){
                cout<<"Found";
            }

            else{
                search(val, current);
            }
            
        }

        


};

int main(){

    binaryTree tree;
    tree.add(2);

    return 0;
}

File: Binary Tree/tree.py
class node:
    def __init__(self, val):
        self.data = val
        self.right = None
        self.left = None


class Tree:

    def __init__(self):
        
        self.root = None

    def move(self, val, node):

        if node.data == val:
            return "Found"
        
        elif node.data < val:
            return node.right
        
        elif node.data > val:
            return node.left


    def insert(self, val):

        newnode = node(val)
        newnode.right = None
        newnode.left = None   

        if self.root == None:
            self.root = newnode

        else:
            current = self.root

            while current != None:

                current = self.move(val, current)
                
            current = newnode
                

    def search(self, val):

        current = self.root

        while current != None:
                
            current = self.move(val, current)
            if current == "Found":
                return "Found"
                
        return "Not Found"



                


tree = Tree()
tree.insert(1)
tree.insert(3)
tree.insert(10)
print(tree.search(10))